source/fundamentals/aggregation.txt

```diff
@@ -0,0 +1,124 @@
+===========
+Aggregation
+===========
+
+.. default-domain:: mongodb
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+Overview
+--------
+
+In this guide, you can learn how to use **aggregation operations** in
+the MongoDB Node.js driver.
+
+Aggregation operations are expressions you can use to produce reduced
+and summarized results in MongoDB. MongoDB's aggregation framework
+allows you to create a pipeline that consists of one or more stages,
+each of which performs a specific operation on your data.
+
+You can think of the aggregation framework as similar to an automobile factory.
+Automobile manufacturing requires the use of assembly stations organized
+into assembly lines. Each station has specialized tools, such as
+drills and welders. The factory transforms and
+assembles the initial parts and materials into finished products.
+
+The **aggregation pipeline** is the assembly line, **aggregation stages** are the assembly stations, and
+**operator expressions** are the specialized tools.
+
+Aggregation vs. Query Operations
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Using query operations, such as the ``find()`` method, you can:
+
+- select *which documents* to return
+- select *which fields* to return
+- sort the results
+
+Using aggregation operations, you can:
+
+- perform all query operations
+- rename fields
+- calculate fields
+- summarize data
+- group values
+
+Aggregation operations have some :manual:`limitations </core/aggregation-pipeline-limits/>`:
+
+- Returned documents must not violate the :manual:`BSON-document size limit </reference/limits/#mongodb-limit-BSON-Document-Size>`
+  of 16 megabytes.
+
+- Pipeline stages have a memory limit of 100 megabytes by default. If necessary, you may exceed this limit by setting the ``allowDiskUse``
+  property of ``AggregateOptions`` to ``true``. See the
+  :node-api-4.0:`AggregateOptions API documentation <interfaces/aggregateoptions.html>`
+  for more details.
+
+.. important:: ``$graphLookup`` exception
+
+   The :manual:`$graphLookup
+   </reference/operator/aggregation/graphLookup/>` stage has a strict
+   memory limit of 100 megabytes and will ignore ``allowDiskUse``.
+
+Useful References
+~~~~~~~~~~~~~~~~~
+
+- :manual:`Expression operators </reference/operator/aggregation/>`
+- :manual:`Aggregation pipeline </core/aggregation-pipeline/>`
+- :manual:`Aggregation stages </meta/aggregation-quick-reference/#stages>`
+- :manual:`Operator expressions </meta/aggregation-quick-reference/#operator-expressions>`
+
+Runnable Examples
+-----------------
+
+The following example inserts sample data into the ``restaurants``
+collection of the ``aggregation`` database.
```

Comment:
```suggestion
The examples use a fictional sample data set about restaurants. The following code inserts the sample data into the ``restaurants``
collection of the ``aggregation`` database.
```

I think this makes it sound like the following code snippet demonstrates aggregation. I think this should be clarified so it is clear that this is only here to create a sample dataset so that aggregation can be demonstrated later.

Reply:
I separated the sentences so that the insertion sentence comes right before the code.